asapdiscovery-alchemy/asapdiscovery/alchemy/utils.py

```diff
@@ -175,3 +176,34 @@ def collect_results(
         )
 
         return network_with_results
+
+    def collect_errors(
+        self,
+        planned_network: FreeEnergyCalculationNetwork,
+    ) -> list[AlchemiscaleFailure]:
+        """
+        Collect errors and tracebacks from failed tasks.
+
+        Args:
+            planned_network: Network to get failed tasks from.
+
+        Returns:
+            List of AlchemiscaleFailure objects with errors and tracebacks for the failed tasks in the network.
+        """
+        network_key = planned_network.results.network_key
+        errored_tasks = self._client.get_network_tasks(network_key, status="error")
+
+        error_data = []
+        for task in errored_tasks:
```

Comment:
> .. would also grabbing the ligand names (and showing them in that box) for the edge be in scope for this PR?

The shortest path to getting ligand names would be, for each `Task` `ScopedKey`, doing `self._client.get_task_transformation(task_sk).name`. With the way [`Transformation`s are built up](https://github.com/choderalab/asapdiscovery/blob/f18e5dfb35328190bc346023c457e0312eebbcd1/asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py#L333-L339), this should give sufficient information to track down which ligands are involved.

Reply:
While this is possible, it gets a bit intricate in terms of having to inspect transformations -> ChemicalSystems -> Components and filter the ligands. I think I prefer to leave this one for improvements in a future PR, if we really end up needing this.

Reply:
I don't think it's as intricate as you're making it in this case. Because we build up the `Transformation` names in a particular way, they already include information on the ligands involved in the failed transformation. There is no need to introspect further down into the `ChemicalSystems`, though you could choose to if you prefer since you would have those already.

Something I'd like to do to make this way more performant is to add methods like `get_transformation_name`, `get_chemicalsystem_name`, etc. for getting back the names of these objects without having to pull down a whole chain of `gufe` objects and put them back together. 

Also, perhaps methods `get_chemicalsystem_components` and `get_component_name` would make it quick to get more granular information without full `gufe` objects as well.

Reply:
can tackle this in a next PR!

Reply:
Yes, I agree, I'm unsure how good it is to rely on the transformation names for this. Maybe that's robust enough, but for the sake of moving forward with this PR I think we could discuss it in its own issue.